### PR TITLE
Plugin for Google meet

### PR DESCRIPTION
**Pull request by aragonc**
Un pequeño plugin para administrar URLs de reuniones Google Meets.

Espero observaciones para mejorarlo.

Saludos.

### CODE REVIEW

File: plugin/googlemeet/resources/css/style.css [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+}
+.row-meet {
+    display: block;
+}
+
+.description{
+    padding-left: 1rem;
+    padding-right: 1rem;
+    padding-top: 2rem;
+    font-size: 12px;
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:43]: Codacy Issue found: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

**Comment by jmontoyaa** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 69
           

Complexity increasing per file
==============================
- plugin/googlemeet/src/googlemeet_plugin.class.php  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281&bid=18611862)

File: plugin/googlemeet/resources/css/style.css [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+.meet-list .meet-item{
+    margin: 10px;
+}
+.meet-list .meet-item .card-meet{
+    min-height: 220px;
+}
+@media (min-width: 768px) and (max-width: 979px) {
+    .meet-list {
+        display: grid;
+        grid-template-columns: repeat(2, 1fr);
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:133]: Codacy Issue found: [Expected RBRACE at line 133, col 42.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    margin-top: 1rem;
+}
+.card{
+    position: relative;
+    display: flex;
+    flex-direction: column;
+    min-width: 0;
+    word-wrap: break-word;
+    border: 1px solid rgba(0,0,0,.05);
+    border-radius: .375rem;
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:20]: Codacy Issue found: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+}
+.card-meet{
+    margin-bottom: 30px;
+    border: 0;
+    box-shadow: 0 0 2rem 0 rgba(136,152,170,.15);
+}
+.card-meet:hover{
+    border-left: .25rem solid #cdcdcd !important;
+}
+.card-meet .card-body {
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:32]: Codacy Issue found: [Expected empty line before rule (rule-empty-line-before)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    margin-bottom: 2rem;
+    margin-top: 1rem;
+}
+.card{
+    position: relative;
+    display: flex;
+    flex-direction: column;
+    min-width: 0;
+    word-wrap: break-word;
+    border: 1px solid rgba(0,0,0,.05);
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:19]: Codacy Issue found: [Expected a leading zero (number-leading-zero)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    padding: 1rem 1.5rem;
+}
+.row-meet {
+    display: block;
+}
+
+.description{
+    padding-left: 1rem;
+    padding-right: 1rem;
+    padding-top: 2rem;
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:42]: Codacy Issue found: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/start.php [hunk 7]
```diff
@@ -0,0 +1,47 @@
+<?php
```
Comment (jmontoyaa) [inline plugin/googlemeet/start.php:1]: Codacy Issue found: [End of line character is invalid; expected "'\n'" but found "\n"](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/install.php [hunk 8]
```diff
@@ -0,0 +1,10 @@
+<?php
```
Comment (jmontoyaa) [inline plugin/googlemeet/install.php:1]: Codacy Issue found: [End of line character is invalid; expected "'\n'" but found "\n"](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+    min-width: 0;
+    word-wrap: break-word;
+    border: 1px solid rgba(0,0,0,.05);
+    border-radius: .375rem;
+    background-color: #fff;
+    background-clip: border-box;
+}
+.card-meet{
+    margin-bottom: 30px;
+    border: 0;
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:26]: Codacy Issue found: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+.description{
+    padding-left: 1rem;
+    padding-right: 1rem;
+    padding-top: 2rem;
+    font-size: 12px;
+    color: #666666;
+    display: grid;
+}
+.description .title{
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:47]: Codacy Issue found: [Expected single space before "{" (block-opening-brace-space-before)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+.card-meet{
+    margin-bottom: 30px;
+    border: 0;
+    box-shadow: 0 0 2rem 0 rgba(136,152,170,.15);
+}
+.card-meet:hover{
+    border-left: .25rem solid #cdcdcd !important;
+}
+.card-meet .card-body {
+    padding: 1rem 1.5rem;
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:33]: Codacy Issue found: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+    display: block;
+}
+
+.description{
+    padding-left: 1rem;
+    padding-right: 1rem;
+    padding-top: 2rem;
+    font-size: 12px;
+    color: #666666;
+    display: grid;
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:45]: Codacy Issue found: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+    background-color: #fff;
+    background-clip: border-box;
+}
+.card-meet{
+    margin-bottom: 30px;
+    border: 0;
+    box-shadow: 0 0 2rem 0 rgba(136,152,170,.15);
+}
+.card-meet:hover{
+    border-left: .25rem solid #cdcdcd !important;
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:30]: Codacy Issue found: [Use of !important](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/config.php [hunk 14]
```diff
@@ -0,0 +1,11 @@
+
```
Comment (jmontoyaa) [inline plugin/googlemeet/config.php:1]: Codacy Issue found: [End of line character is invalid; expected "'\n'" but found "\n"](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/lang/spanish.php [hunk 15]
```diff
@@ -0,0 +1,26 @@
+<?php
+/* For licensing terms, see /license.txt */
+
+$strings['plugin_title'] = 'Hangouts Meet';
+$strings['plugin_comment'] = 'Google Hangouts Meet es el sistema de videoconferencia de Google, Conformado por a la app de Hangouts Meet y el Hardware de videoconferencia.';
+$strings['tool_title'] = 'Nombre de la herramienta';
+$strings['tool_title_help'] = 'Puede cambiar el nombre de la herramienta Google Meet; Ejemplo: Videoconferencia';
+$strings['google_meet_enabled'] = 'Activar Herramienta Google Meet';
+$strings['google_meet_enabled_help'] = 'Escoga si desea activar el plugin en su plataforma Chamilo, esta acción activara una herramienta extra en las herramienta del curso.';
```
Comment (jmontoyaa) [inline plugin/googlemeet/lang/spanish.php:9]: Codacy Issue found: [Line exceeds maximum limit of 120 characters; contains 174 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css [hunk 16]
```diff
@@ -0,0 +1,148 @@
+.plugin_logo{
+    text-align: center;
+    padding-top: 1rem;
+    padding-bottom: 1rem;
+}
+.plugin_logo .img-responsive{
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:6]: Codacy Issue found: [Expected single space before "{" (block-opening-brace-space-before)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/src/googlemeet_plugin.class.php [hunk 17]
```diff
@@ -0,0 +1,297 @@
+<?php
```
Comment (jmontoyaa) [inline plugin/googlemeet/src/googlemeet_plugin.class.php:1]: Codacy Issue found: [End of line character is invalid; expected "'\n'" but found "\n"](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css [hunk 18]
```diff
@@ -0,0 +1,148 @@
...
+    text-overflow: ellipsis;
+    width: 100%;
+    display: inline-block;
+}
+.description .text{
+    display: grid;
+    padding-bottom: 1rem;
+}
+.rounded-circle {
+    border-radius: 50%!important;
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:62]: Codacy Issue found: [Use of !important](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css (hunk 5 above)
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:19]: Codacy Issue found: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css [hunk 19]
```diff
@@ -0,0 +1,148 @@
...
+    word-wrap: break-word;
+    border: 1px solid rgba(0,0,0,.05);
+    border-radius: .375rem;
+    background-color: #fff;
+    background-clip: border-box;
+}
+.card-meet{
+    margin-bottom: 30px;
+    border: 0;
+    box-shadow: 0 0 2rem 0 rgba(136,152,170,.15);
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:27]: Codacy Issue found: [Expected single space after "," in a single-line function (function-comma-space-after)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css [hunk 20]
```diff
@@ -0,0 +1,148 @@
+.plugin_logo{
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:1]: Codacy Issue found: [Expected single space before "{" (block-opening-brace-space-before)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/uninstall.php [hunk 21]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (jmontoyaa) [inline plugin/googlemeet/uninstall.php:1]: Codacy Issue found: [End of line character is invalid; expected "'\n'" but found "\n"](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css [hunk 22]
```diff
@@ -0,0 +1,148 @@
...
+    text-align: center;
+    padding-top: 1rem;
+    padding-bottom: 1rem;
+}
+.plugin_logo .img-responsive{
+    margin: auto;
+}
+.tools{
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:9]: Codacy Issue found: [Expected empty line before rule (rule-empty-line-before)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:9]: Codacy Issue found: [Expected single space before "{" (block-opening-brace-space-before)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

File: plugin/googlemeet/resources/css/style.css [hunk 23]
```diff
@@ -0,0 +1,148 @@
...
+.card{
+    position: relative;
+    display: flex;
+    flex-direction: column;
+    min-width: 0;
+    word-wrap: break-word;
+    border: 1px solid rgba(0,0,0,.05);
+    border-radius: .375rem;
+    background-color: #fff;
+    background-clip: border-box;
```
Comment (jmontoyaa) [inline plugin/googlemeet/resources/css/style.css:22]: Codacy Issue found: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5686281)

**Comment by ywarnier** [comment]
Unas observaciones para @aragonc 
* si existe un plugin google_maps, llama el tuyo google_meet y no googlemeet. El hecho que Chamilo no lo tome de inmediato es una cuestión de crear un método "getName()" dentro de la clase de tu plugin que devuelva el buen nombre (google_meet) en lugar de intentar construirlo en base al nombre de la clase
* filtrar siempre las cosas que envías a Database::query(). Database::insert(), update() y delete() ya tienen filtrado, pero query() no, así que dejándo variable sin filtrar, estás creando una falla de seguridad para Chamilo
* en todo Chamilo hemos usado siempre c_id, y tu creas cd_id. Hasta de ti me sorprendió :-)
* si puedes usar una variable de path que incluya "plugin/", mejor que _p.web + '/plugin/...'.
* más documentación PHPDoc en las funciones, por favor

Por lo general es un buen trabajo. Te felicito y te agradezco.

**Pull request merged**